core/src/utilities/random.cc

```diff
@@ -485,6 +608,34 @@ namespace jiminy
         return 2.0 * grad * delta;
     }
 
+    double PeriodicPerlinNoiseOctave::grad2D(
+        int32_t ix, int32_t iy, double x, double y) const noexcept
+    {
+        // Compute random gradient
+        // Get knot
+        uint64_t knot = (static_cast<uint64_t>(ix) << 32) | static_cast<uint32_t>(iy);
```

Comment:
I'm not convinced by this. The periodicity is supposed to have the spatial interpretation in physical world. A repeating pattern is supposed to be clearly visible. I'm wondering what it looks like in this case. In the case of periodic Perlin noise, I have the feeling that `sumHeightmaps` is the way to go. Anyway, I don't think it really makes sense for ground profile generation.

Reply:
One option would be to do:
```cpp
uint32_t deltaPeriod = static_cast<uint32_t>(period_ / wavelength_)
ix %= deltaPeriod;
iy %= deltaPeriod;
uint64_t knot = (static_cast<uint64_t>(ix) << 32) | static_cast<uint32_t>(iy);
const double s = perm_[knot] / 256.0;
```

By the way, I think there was an issue in the 1D case if `deltaPeriod > 256`